File: Fortran/ex5/featureNormalize.f90
subroutine featureNormalize(M, N, X, mu, sigma)

	integer:: i, j, M, N
	double precision:: X(M,N), mu(N), sigma(N)
	
	call meanN(M, N, X, mu)

	call bsxfunM(M, N, X, mu)
	
			
	call std(M, N, X, sigma)	

	call bsxfunRD(M, N, X, sigma)
	

end subroutine  featureNormalize

File: C/ex5/featureNormalize.c
#include <math.h>
#include <stdio.h>

double meanN(int M, int N, double X[M][N], double mu[N]);
void std(int M, int N, double X[M][N], double std[N]);
void bsxfunM(int M, int N, double X[M][N], double y[M]);
void bsxfunRD(int M, int N, double X[M][N], double y[M]);

int featureNormalize(int M, int N, double X[M][N], double mu[N], double sigma[N])
{
	int i, j;
	
	meanN(M, N, X, mu);

	bsxfunM(M, N, X, mu);			
	std(M, N, X, sigma);		
			
	bsxfunRD(M, N, X, sigma);
	
	
	return 0;
}
